workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -353,38 +365,39 @@ def _get_prescriber_or_ml_stepper(
         stepper_config: Union[PrescriberConfig, MachineLearningConfig],
         step: str,
         hydrostatic: bool = False,
-    ) -> Union[PureMLStepper, Prescriber]:
+    ) -> Union[PureMLStepper, IntervalStepper, Prescriber]:
         if isinstance(stepper_config, MachineLearningConfig):
             model = self._open_model(stepper_config)
-            stepper: Union[PureMLStepper, Prescriber] = PureMLStepper(
+            self._log_info(f"Using PureMLStepper at {step}.")
+            return PureMLStepper(
                 model=model, timestep=self._timestep, hydrostatic=hydrostatic,
             )
-            self._log_info(f"Using PureMLStepper at {step}.")
+
         else:
-            stepper = runtime.factories.get_prescriber(
-                stepper_config, self._get_communicator()
-            )
             self._log_info(
                 "Using Prescriber for variables "
                 f"{list(stepper_config.variables.values())} at {step}."
             )
-        return stepper
+            return runtime.factories.get_prescriber(
```

Comment:
Lots of typing complaints when the returned type varies depending on the if statement